### PR TITLE
update bcftools/annotate

### PR DESCRIPTION
**Pull request by ramprasadn**
## PR checklist

This PR updates the module so it can accept an annotations file and an header-lines file, and it also adds a stub section. 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/bcftools/annotate/main.nf [hunk 1]
```diff
@@ -8,29 +8,33 @@ process BCFTOOLS_ANNOTATE {
...
-    def output_type_compressed = matcher ? "z" : "b"
+    def args    = task.ext.args ?: ''
+    def prefix  = task.ext.prefix ?: "${meta.id}"
+    def header_file = header_lines ? "--header-lines ${header_lines}" : ''
+    def annotations_file = annotations ? "--annotations ${annotations}" : ''
+    def extension = args.contains("--output-type b") || args.contains("-Ob") ? "bcf.gz" :
+                    args.contains("--output-type u") || args.contains("-Ou") ? "bcf" :
+                    args.contains("--output-type z") || args.contains("-Oz") ? "vcf.gz" :
+                    args.contains("--output-type v") || args.contains("-Ov") ? "vcf" :
+                    "vcf.gz"
```
Comment (nvnieuwk) [inline modules/nf-core/bcftools/annotate/main.nf:30]:
```suggestion
                    "vcf"
```
The default extension is plain vcf :)

File: modules/nf-core/bcftools/annotate/main.nf [hunk 2]
```diff
@@ -8,29 +8,33 @@ process BCFTOOLS_ANNOTATE {
...
+    tuple val(meta), path("*.{vcf,vcf.gz,bcf,bcf.gz}"), emit: vcf
+    path "versions.yml"                               , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
-    def args = task.ext.args ?: ''
-    def prefix = task.ext.prefix ?: "${meta.id}"
-
```
Comment (nvnieuwk) [inline modules/nf-core/bcftools/annotate/main.nf:24]: Can you also add a file name collision check? (see [here](https://nf-co.re/docs/contributing/modules#naming-conventions) part 6 for an example)

**Review by nvnieuwk — CHANGES_REQUESTED** [review]
Some small comments but overall very good

File: modules/nf-core/bcftools/annotate/main.nf (hunk 1 above)
Comment (ramprasadn) [reply modules/nf-core/bcftools/annotate/main.nf]: Fixed 👍🏻

File: modules/nf-core/bcftools/annotate/main.nf (hunk 2 above)
Comment (ramprasadn) [reply modules/nf-core/bcftools/annotate/main.nf:24]: Nice! Added 👍🏻

**Pull request merged**